src/rust/bitbox02-rust/src/apps/bitcoin/keypath.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright 2020 Shift Crypto AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::pb;
+pub use pb::btc_script_config::multisig::ScriptType as MultisigScriptType;
+
+use util::bip32::HARDENED;
+
+const BIP44_ACCOUNT_MIN: u32 = HARDENED;
+const BIP44_ACCOUNT_MAX: u32 = HARDENED + 99; // 100 accounts
+
+const PURPOSE_MULTISIG: u32 = 48 + HARDENED;
+const MULTISIG_SCRIPT_TYPE_P2WSH: u32 = 2 + HARDENED;
+const MULTISIG_SCRIPT_TYPE_P2WSH_P2SH: u32 = 1 + HARDENED;
+
+/// Validates a keypath to be
+/// m/expected_purpose/expected_coin/account, where account between 0' and 99'.
+pub fn validate_account(
```

Comment:
Whenever I see "validate_x" functions I think about https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/

Reply:
I like that post a lot, I was aware of it. 

I am not sure how to apply it here though. Ideas?

Reply:
I can't really find the call-site for `validate_account_multisig`. But maybe you could return something like `struct Bip48Account` instead of `()` and thereby using the type system to enforce that you actually did the validation you intended?

Since you already were aware of it, maybe it doesn't fit so well here.

Reply:
Hmm, maybe this'll have to wait until more code is in rust actually.

Reply:
I'll keep it in mind.